.github/workflows/deploy-lambda.yml

```diff
@@ -0,0 +1,172 @@
+    with:
+      ref: ${{ inputs.ab2dLambdasBranch }}
+    if: ${{ inputs.runBuildJob == true }}
+
+  deploy:
+    runs-on: self-hosted
+    needs: publish_lambdas
+    # always run this job but if runBuildJob=true, then wait for 'publish_lambdas' job to finish
+    # without this conditional, this job will not run unless runBuildJob=true
+    if: ${{ always() && !cancelled() }}
+
+    env:
+      ACTIONS_ALLOW_USE_UNSECURE_NODE_VERSION: true
+      AWS_REGION: ${{ vars.AWS_REGION }}
+      DEPLOYMENT_ENV: ${{ vars[format('{0}_DEPLOYMENT_ENV', inputs.environment)] }}
+
+    steps:
+      - name: Set environment variables
+        uses: cmsgov/ab2d-bcda-dpc-platform/actions/aws-params-env-action@main
+        env:
+          AWS_REGION: ${{ vars.AWS_REGION }}
+        with:
+          params: |
+            ARTIFACTORY_URL=/artifactory/url
+            ARTIFACTORY_USER=/artifactory/user
+            ARTIFACTORY_PASSWORD=/artifactory/password
+            OPS_GITHUB_TOKEN=/ci/github/token
+
+      - name: Checkout Code
+        uses: actions/checkout@v3
+        with:
+          repository: 'cmsgov/ab2d-ops'
+          ref: ${{ inputs.ab2dOpsBranch }}
+          token: ${{ env.OPS_GITHUB_TOKEN }}
+
+      - name: Set variables for build version and AB2D environment
+        run: |
+          BUILD_VERSION=$(echo "${{ needs.publish_lambdas.outputs.build_version || inputs.buildVersion }}" | tr -d ' ')
+          echo "BUILD_VERSION=${BUILD_VERSION}"
+          if [ -z "${BUILD_VERSION}" ]; then
+            echo "Unable to determine build version"
+            exit 1
+          fi  
+          
+          if [[ ${{ inputs.environment }} = "dev" ]];         then AB2D_ENV='ab2d-dev'
+          elif [[ ${{ inputs.environment }} = "test" ]];      then AB2D_ENV='ab2d-east-impl'
+          elif [[ ${{ inputs.environment }} = "sbx" ]];       then AB2D_ENV='ab2d-sbx-sandbox'
+          elif [[ ${{ inputs.environment }} = "prod_test" ]]; then AB2D_ENV='ab2d-east-prod-test'
+          elif [[ ${{ inputs.environment }} = "prod" ]];      then AB2D_ENV='ab2d-east-prod'
+          else echo "Unable to determine AB2D environment"; exit 1
+          fi
+          
+          echo "BUILD_VERSION=${BUILD_VERSION}" >> $GITHUB_ENV
+          echo "AB2D_ENV=${AB2D_ENV}" >> $GITHUB_ENV
+
+      - name: Assume role in AB2D account for this environment
+        uses: aws-actions/configure-aws-credentials@v3
+        env:
+          ACCOUNT: ${{ inputs.environment == 'prod_test' && 'prod' || inputs.environment }}
+        with:
+          aws-region: ${{ vars.AWS_REGION }}
+          role-to-assume: arn:aws:iam::${{ secrets[format('{0}_ACCOUNT_ID', env.ACCOUNT)] }}:role/delegatedadmin/developer/ab2d-${{ env.ACCOUNT }}-github-actions
+
+      - name: Download ZIP files
+        env:
+          ARTIFACTORY_PREFIX: 'https://artifactory.cloud.cms.gov/artifactory/ab2d-main/gov/cms/ab2d'
+        run: |
+          mkdir -p terraform/modules/lambda/event-service || true
```

Comment:
No need for ` || true` here as `mkdir -p` will not error on the directory already existing or path pieces missing.